### DISCUSSION TITLE
Request: Option to override tiling accessibility actions on a per app basis

### DISCUSSION THREAD

**Discussion by mcandre** [rxhanson/Rectangle#1486] [Ideas]
Rectangle and its paid cousin Rectangle Pro both suffer from integration quirks with VLC. For example, attempts to maximize VLC via Rectangle result in the VLC video expanding beyond a proper Fit to Screen, with the bottom of the video dangling behind the macOS Dock.

Previously, it was noted that VLC may not have the best macOS accessibility API integration. Be that as it may, there should still be a way to effectively get the Rectangle maximize hotkey to forward to VLC menu hotkeys. That would provide a practical workaround.

VLC offers its own menu hotkey action Fit to Screen in the Video menubar section. The default hotkey for this action is Command+3.

macOS allows overriding the hotkeys associated with menubar actions, in System Settings -> Keyboard -> Keyboard Shortcuts...

However, we then have a sub-problem of a collision between Rectangle's maximize tiling hotkey and VLC's Fit to Screen maximize hotkey.

Fortuantely, we have two coding strategies we can try to resolve the integration problems, and they're not mutually exclusive.

First, it may prove helpful for Rectangle to provide a per-app option, to unregister the hotkeys for a given list of tiling actions whenever the app is foregrounded. Then, when a UI element not from the app is foregrounded, Rectangle could reregister the tiling hotkey. That's one way to allow the same hotkey to quietly pass through to more complicated applications.

Another strategy involves Rectangle providing a per-app option to replace tiling actions with forwarding custom hotkey combinations. This method has the benefit of not having to override VLC's Fit to Screen default hotkey; Rectangle could learn to simply replace the maximize accessibility API call with VLC's Fit to Screen hotkey / an arbitrary app hotkey of the user's choice. As a bonus, this strategy gathers the entire system configuration into the Rectangle app, making it much simpler to quickly provision the desired settings onto additional computers via Rectangle configuration import & export.

Until at least one of these enhancements is implemented, then I have to terminate Rectangle with `killall Rectangle` (I disable the menulet, and I don't see a button in the Rectangle management UI to quit the program.) Unfortunately, that destroys productivity.

When Rectangle does integrate well with other applications, I feel rather productive. I wish Rectangle would sort out VLC problems much faster. Previously, Rectangle (and grandpappy ShiftIt) were a major reason I stuck with macOS at all.

I had hoped that Rectangle Pro would include these kinds of workarounds to integrate better with VLC, but I was unable to find such options. So there is nothing available in Rectangle Pro that would compel me to purchase it.

**Comment by rxhanson** [comment]
Did you try ignoring VLC in Rectangle/Pro? When VLC is frontmost, in the Rectangle/Pro menu bar menu select Ignore VLC. Let me know if that's not what you're looking for.

**Reply by mcandre** [reply]
Fascinating to hear such options.

I was under the impression that 100% of the Rectangle [Pro] settings were adjustable within the Rectangle [Pro] settings panel. If some settings are managed elsewhere, that's neat but not exactly intuitive.

**Reply by mcandre** [reply]
By the way, VLC and other macOS applications often present so many menu items, that the Rectangle menulet icon disappears.

Would be helpful to present similar controls in the ordinary Rectangle settings view.

**Reply by rxhanson** [reply]
I see what you mean. I'll give this some thought for a future update, maybe as a library that I can share to provide this functionality in other apps.

**Reply by mcandre** [reply]
as a library eh?

that's the dream
